cv-server/src/MockerWebcamREST/mocker_webcam.py

```diff
@@ -0,0 +1,241 @@
+            font = helv36
+        )
+        self.play_btn = tk.Button(
+            self.bottom_frame,
+            text = 'PLAY',
+            command = self.play_video,
+            font = helv36
+        )
+        self.stop_btn = tk.Button(
+            self.bottom_frame,
+            text = 'STOP',
+            command = self.stop_video,
+            font = helv36
+        )
+        self.replay_btn = tk.Button(
+            self.bottom_frame,
+            text = 'REPLAY',
+            command = self.replay_video,
+            font = helv36
+        )
+        self.video_canvas = tk.Canvas(
+            self.top_frame,
+        )
+        self.video_canvas.grid(row=0, column=0)
+        self.toggle_webcam_btn.grid(row=0, column=0, padx=5, pady=5, ipadx=10)
+        self.open_file_btn.grid(row=0, column=1, padx=5, pady=5, ipadx=10)
+        self.replay_btn.grid(row=0, column=2, padx=5, pady=5, ipadx=10)
+        self.play_btn.grid(row=0, column=3, padx=5, pady=5, ipadx=10)
+        self.stop_btn.grid(row=0, column=4, padx=5, pady=5, ipadx=10)
+
+
+    def update(self):
+        if self.video_capture.video != None and self.playing:
+            ret, frame = self.video_capture.get_frame()
+            self.ret = ret
+            if self.ret:
+                self.current_frame = frame
+                self.photo = PIL.ImageTk.PhotoImage(image = PIL.Image.fromarray(frame))
+                self.video_canvas.create_image(0,0, image = self.photo, anchor = tk.NW)
+            else:
+                self.current_frame = None
+                self.playing = False
+        self.after(self.delay, self.update)
+
+    def toggle_webcam(self):
+        if self.webcam:
+            self.toggle_webcam_btn.config(background = "#7E7E7E")
+            self.replay_btn["state"] = "normal"
+            self.open_file_btn["state"] = "normal"
+            self.play_btn["state"] = "normal"
+            self.stop_btn["state"] = "normal"
+            self.video_capture.reset()
+            self.webcam = False
+        else:
+            self.toggle_webcam_btn.config(background = "red")
+            self.replay_btn["state"] = "disabled"
+            self.open_file_btn["state"] = "disabled"
+            self.play_btn["state"] = "disabled"
+            self.stop_btn["state"] = "disabled"
+            self.video_capture.set(0)
+            self.resize_video_canvas()
+            self.playing = True
+            self.webcam = True
+
+    def play_video(self):
+        if self.video_capture.video != None:
+            self.playing = True
+    
+    def stop_video(self):
+        if self.video_capture.video != None:
+            self.playing = False
+
+    def replay_video(self):
+        if self.video_capture.video != None:
+            self.playing = True
+            self.video_capture.set(self.video_capture.current)
+
+    def select_file(self):
+        filename = filedialog.askopenfile(
+            title='Open a video file',
+            initialdir = os.getcwd(),
```

Comment:
W metodach i funkcja jak ustawiamy jakiś parametr wyjściowy
method(param=my_param)
to nie dajemy spacji przed i po '='

ale to taki szkopuł